src/_components/Assets/AddAssetContainer.jsx

```diff
@@ -156,17 +186,31 @@ class AddAssetContainer extends React.Component {
           onAddAssetTag={this.onAddAssetTag}
           onCreateAsset={this.onCreateAsset}
           goBack={this.goBack}
-          filteredSubCategories={this.state.filteredSubCategories}
-          filteredAssetTypes={this.state.filteredAssetTypes}
-          filteredAssetMakes={this.state.filteredAssetMakes}
           filteredModelNumbers={this.state.filteredModelNumbers}
           modelNumber={this.state.modelNumber}
           serialNumber={this.state.serialNumber}
           assetTag={this.state.assetTag}
           buttonState={this.state.saveButtonState}
           onChangeButtonState={this.onChangeButtonState}
+          year={this.state.year}
+          processorType={this.state.processorType}
+          processorSpeed={this.state.processorSpeed}
+          screenSize={this.state.screenSize}
+          storage={this.state.storage}
+          memory={this.state.memory}
+          onSelectYearOfManufacture={this.onSelectYearOfManufacture}
+          onSelectProcessorType={this.onSelectProcessorType}
+          onSelectProcessorSpeed={this.onSelectProcessorSpeed}
+          onSelectScreenSize={this.onSelectScreenSize}
+          onSelectStorage={this.onSelectStorage}
+          onSelectMemory={this.onSelectMemory}
         /> : <FilterAssetComponent
           {...this.props}
+          handleDropdownChanges={this.handleDropdownChanges}
```

Comment:
The render function of this component is becoming unwieldy due to the ternary operator and numerous props. Have you considered if things can be broken down to smaller components? For starters, use an if instead of the ternary operator. I.e.
```
render() {
  if (this.state.page === 1) {
    return (
      <AddAssetComponent
      ...
      ...
      />
    );
  }

  return (
    <FilterAssetComponent
      ...
      ...
    />
  );
}
```

Reply:
@zmwas and I had the same concern you are raising , we are looking into a more efficient way of doing this.

Reply:
@sachungo we have refactored the add asset component. It now has SpecComponent as a child and the props are passed as an object.